src/pepr/operator/controllers/network/authorizationPolicies.ts

```diff
@@ -0,0 +1,272 @@
+}
+
+/**
+ * Helper to build an AuthorizationPolicy from rule details.
+ * If the computed source is empty, the "from" field is omitted.
+ */
+function buildAuthPolicy(
+  policyName: string,
+  pkg: UDSPackage,
+  selector: Record<string, string> | undefined,
+  source: Source,
+  ports: string[],
+): AuthorizationPolicy {
+  const ruleEntry: Rule = {};
+  if (!isEmpty(source)) {
+    ruleEntry.from = [{ source }];
+  }
+  if (ports.length > 0) {
+    ruleEntry.to = [{ operation: { ports } }];
+  }
+
+  const pkgName = pkg.metadata?.name ?? "unknown";
+  const pkgNamespace = pkg.metadata?.namespace ?? "default";
+  const generation = pkg.metadata?.generation?.toString() ?? "0";
+
+  return {
+    apiVersion: "security.istio.io/v1beta1",
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: policyName,
+      namespace: pkgNamespace,
+      labels: {
+        "uds/package": pkgName,
+        "uds/generation": generation,
+        "uds/for": "network",
+      },
+      ownerReferences: getOwnerRef(pkg),
+    },
+    spec: {
+      action: Action.Allow,
+      ...(selector ? { selector: { matchLabels: selector } } : {}),
+      rules: [ruleEntry],
+    },
+  };
+}
+
+/**
+ * Generate and apply Istio Authorization Policies for a given UDSPackage.
+ */
+export async function generateAuthorizationPolicies(
+  pkg: UDSPackage,
+  pkgNamespace: string,
+): Promise<AuthorizationPolicy[]> {
+  const pkgName = pkg.metadata?.name ?? "unknown";
+  const generation = pkg.metadata?.generation?.toString() ?? "0";
+  log.info(
+    `Starting policy generation for package "${pkgName}" in namespace "${pkgNamespace}" (generation ${generation}).`,
+  );
+
+  const policies: AuthorizationPolicy[] = [];
+
+  // Process allow rules.
+  if (pkg.spec?.network?.allow) {
+    for (const rule of pkg.spec.network.allow) {
+      if (rule.direction === "Egress") continue;
+      const { source, ports } = processAllowRule(rule, pkgNamespace);
+      const policyName = sanitizeResourceName(`protect-${pkgName}-${generateAllowName(rule)}`);
+      const authPolicy = buildAuthPolicy(policyName, pkg, rule.selector, source, ports);
+      policies.push(authPolicy);
+      log.debug(`Generated authpol: ${authPolicy.metadata?.name}`);
+    }
+  }
+
+  // Process expose rules.
+  if (pkg.spec?.network?.expose) {
+    for (const rule of pkg.spec.network.expose) {
+      const { source, ports } = processExposeRule(rule);
+      const policyName = sanitizeResourceName(`protect-${pkgName}-${generateExposeName(rule)}`);
+      const authPolicy = buildAuthPolicy(policyName, pkg, rule.selector, source, ports);
+      policies.push(authPolicy);
+      log.debug(`Generated authpol: ${authPolicy.metadata?.name}`);
+    }
+  }
+
+  // Process monitor rules.
+  if (pkg.spec?.monitor) {
+    for (const monitor of pkg.spec.monitor) {
+      const selector = monitor.podSelector ?? monitor.selector;
+      const source: Source = { namespaces: ["monitoring"] };
```

Comment:
Source here should be `principal`, with the prometheus service account (to be most specific/secure).